lib/characteristic.dart

```diff
@@ -69,7 +69,7 @@ class Characteristic extends InternalCharacteristic {
   /// The value can be written only if [isWritableWithResponse] or
   /// [isWritableWithoutResponse] is `true` and argument [withResponse] is
   /// set accordingly.
-  Future<void> write(
+  Future<Characteristic?> write(
```

Comment:
Void here was intentional. There's a bug with not returning response from write with response, but it has to be fixed on multiple layers and even then this would be typed as `Uint8List`.